src/main/kotlin/ResponseBuilder.kt

```diff
@@ -1,16 +1,17 @@
 package response
 
-class ResponseBuilder(private val statusCode: Int, private val body: String = "") {
+class ResponseBuilder(private val statusCode: Int,
+                      private val body: String = "",
+                      private val headers: Map<String, List<String>> = mapOf()
+                      ) {
     private val protocol = "HTTP/1.1"
     private val crlf = "\r\n"
 
     fun build(): String {
         val responseLine = generateResponseLine()
-        if (body.isNotEmpty()) {
-            val contentLength = setContentLength()
-            return responseLine + "$contentLength$crlf$crlf$body"
-        }
-        return responseLine
+        var responseHeaders = generateHeaders()
+        var responseBody = if (body.isNotEmpty()) setContentLength() + "$crlf$body" else ""
+        return responseLine + responseHeaders + responseBody
     }
 
     private fun generateResponseLine(): String {
```

Comment:
nitpick: Looks like the first line is called a "status line". The name "responseLine" was a bit confusing. I wasn't sure what that meant. I'd recommend changing this to `generateStatusLine`.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html